### PR TITLE
automount: if an automount unit is masked, don't react to activation …

### PR DESCRIPTION
**Pull request by poettering**
…anymore

Otherwise we'll hit an assert sooner or later.

Fixes: #5441

### CODE REVIEW

**Comment by mbiebl** [comment]
systemd no longer crashes with this patch applied.
Instead, when trying to access /proc/sys/fs/binfmt_misc/, I get
```
Feb 24 19:49:40 debian systemd[1]: proc-sys-fs-binfmt_misc.automount: Got automount request for (null), triggered by 505 (ls)
Feb 24 19:49:40 debian systemd[1]: proc-sys-fs-binfmt_misc.automount: Unit entered failed state.
Feb 24 19:49:46 debian systemd[1]: proc-sys-fs-binfmt_misc.automount: Cannot add dependency job, ignoring: Unit proc-sys-fs-binfmt_misc.automount is masked.
```

The ```ls -l /proc/sys/fs/binfmt_process/``` process simply hangs after that.

File: src/core/automount.c [hunk 1]
```diff
@@ -733,6 +733,12 @@ static void automount_enter_runnning(Automount *a) {
 
         assert(a);
 
+        /* If the user masked our unit in the meantime, fail */
+        if (UNIT(a)->load_state != UNIT_LOADED) {
+                log_unit_debug(UNIT(a), "Suppressing automount request since unit is not longer loaded.");
```
Comment (mbiebl) [inline src/core/automount.c]: "no longer" ?

**Comment by poettering** [comment]
I force pushed a new version now: it accepts automount tokens even if masked, but will then fail them immediately. Effectively this means if you try to trigger a masked automount you'll get an EHOSTDOWN error immediately, the automount unit will then enter fail state (and thus be unmounted). If you you then try to access the mount again it will be gone, hence you'll see the underlying directory instead.

To me this appears pretty OK behaviour now.

**Comment by mbiebl** [comment]
The new behaviour is much nicer. Trying to access the path from the masked automount unit properly marks the automount unit as failed.
The ls process no longer hangs but you get a ```ls: cannot access '/proc/sys/fs/binfmt_misc/': Host is down```

Thanks @poettering

**Pull request merged**